File: src/Java/src/BinarySearch.java
import java.util.Arrays;

public class BinarySearch {
  public static int binarySearch(int[] array, int target) {
    if (array.length != 0) {
      final int mid = (int) Math.floor(array.length / 2);
      if (array[mid] != target) {
        if (array[mid] > target) {
          return binarySearch(Arrays.copyOfRange(array, 0, mid), target);
        }
        int result = binarySearch(Arrays.copyOfRange(array, mid, array.length), target);
        if (result != -1) {
          return mid + result;
        }
        return result;
      }
      return mid;
    }
    return -1;
  }
}

File: src/Python/src/binary_search.py
import math
def binary_search(list: list[int], target: int) -> int:
  if (len(list) != 0):
    MID = math.floor(len(list) / 2)
    GUESS = list[MID]
    if (GUESS != target):
      if (GUESS > target): return binary_search(list[0: MID], target)
      RESULT = binary_search(list[MID:], target)
      if (RESULT != -1):
        RESULT += MID;
      return RESULT  
    return MID
  return -1